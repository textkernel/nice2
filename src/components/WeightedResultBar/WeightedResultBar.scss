.WeightedResultBar {
    + .WeightedResultBar {
        margin-top: var(--spacing-4x);
    }

    &__details {
        margin-bottom: var(--spacing-2x);
        display: flex;
        justify-content: space-between;
    }
}
