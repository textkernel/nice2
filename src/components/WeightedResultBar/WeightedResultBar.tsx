import * as React from 'react';
import { Context } from '../../constants';

interface Props extends React.HTMLAttributes<HTMLDivElement> {
    /** Is component data still loading */
    isLoading?: boolean;
    /** Description of the result */
    children?: string;
    /** Color context for the weighted bar */
    context?: Context;
    /** Weight of this result in percentage */
    percentage?: number;
    /** Weight of this result in absolute number */
    count?: number;
    /** Callback to be executed if component is clicked */
    onClick?: () => void;
}

export const WeightedResultBar: React.FC<Props> = (props) => {
    const {
        isLoading = false,
        children = '',
        context = 'brand',
        percentage = 100,
        count = 0,
        ...rest
    } = props;

    return <div {...rest}>{children}</div>;
};

WeightedResultBar.displayName = 'WeightedResultBar';
