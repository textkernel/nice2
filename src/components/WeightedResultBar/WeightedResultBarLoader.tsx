import * as React from 'react';
import { bem } from '../../utils';
import { ContentPlaceholder } from '../ContentPlaceholder';
import { ProgressBar } from '../ProgressBar';
import styles from './WeightedResultBar.scss';

const { block } = bem('WeightedResultBar', styles);

export const WeightedResultBarLoader: React.FC<React.HTMLAttributes<HTMLDivElement>> = (props) => {
    const [percentageToShow, setPercentageToShow] = React.useState(0);

    // simulate prop change on ProgressBar to get a little animation going
    React.useEffect(() => {
        setTimeout(() => {
            setPercentageToShow(100);
        }, 0);
    }, []);

    return (
        <div {...props} {...block(props)}>
            <ContentPlaceholder height={17} width={Math.floor(Math.random() * 60) + 25} />
            <ProgressBar percentage={percentageToShow} context="neutral" small />
        </div>
    );
};

WeightedResultBarLoader.displayName = 'WeightedResultBarLoader';
