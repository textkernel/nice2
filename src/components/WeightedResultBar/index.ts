export { WeightedResultBar } from './WeightedResultBar';
export { WeightedResultBarLoader } from './WeightedResultBarLoader';
