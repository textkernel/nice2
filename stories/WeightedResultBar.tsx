import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number } from '@storybook/addon-knobs';
import { WeightedResultBar, WeightedResultBarLoader, Button } from '@textkernel/oneui';

storiesOf('Molecules|WeightedResultBar', module)
    .addDecorator(withKnobs)
    .add('WeightedResultBar', () => {
        return (
            <div style={{ width: 500 }}>
                <WeightedResultBar
                    percentage={number('Percentage', 100)}
                    count={number('Count', 123)}
                    context="primary"
                >
                    Repair and Maintenance Technician
                </WeightedResultBar>
                <WeightedResultBar percentage={76} count={94}>
                    Sales Manager
                </WeightedResultBar>
                <WeightedResultBar percentage={64} count={79}>
                    Software Engineer
                </WeightedResultBar>
                <WeightedResultBar percentage={64} count={79}>
                    Operations Manager
                </WeightedResultBar>
                <WeightedResultBar percentage={58} count={72}>
                    Business Development Manager
                </WeightedResultBar>
            </div>
        );
    })
    .add('With a custom count', () => {
        const ClickableCount = ({ count }) => (
            <Button
                size="small"
                context="brand"
                onClick={() => console.log(`WeightedResultBar count: ${count}`)}
            >
                {count}
            </Button>
        );

        return (
            <div style={{ width: 500 }}>
                <WeightedResultBar
                    percentage={number('Percentage', 100)}
                    count={<ClickableCount count={number('Percentage', 100)} />}
                    context="primary"
                >
                    Repair and Maintenance Technician
                </WeightedResultBar>
                <WeightedResultBar percentage={76} count={<ClickableCount count={94} />}>
                    Sales Manager
                </WeightedResultBar>
                <WeightedResultBar percentage={64} count={<ClickableCount count={79} />}>
                    Software Engineer
                </WeightedResultBar>
                <WeightedResultBar percentage={64} count={<ClickableCount count={79} />}>
                    Operations Manager
                </WeightedResultBar>
                <WeightedResultBar percentage={58} count={<ClickableCount count={72} />}>
                    Business Development Manager
                </WeightedResultBar>
            </div>
        );
    })
    .add('WeightedResultBarLoader', () => {
        return (
            <div style={{ width: 500 }}>
                <WeightedResultBarLoader />
                <WeightedResultBarLoader />
                <WeightedResultBarLoader />
                <WeightedResultBarLoader />
                <WeightedResultBarLoader />
            </div>
        );
    });
