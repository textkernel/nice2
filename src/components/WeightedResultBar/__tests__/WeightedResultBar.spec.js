import React from 'react';
import toJson from 'enzyme-to-json';
import { WeightedResultBar } from '../WeightedResultBar';

describe('WeightedResultBar', () => {
    it('should render correctly', () => {
        const wrapper = mount(
            <WeightedResultBar percentage={67} count={123}>
                Result
            </WeightedResultBar>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
