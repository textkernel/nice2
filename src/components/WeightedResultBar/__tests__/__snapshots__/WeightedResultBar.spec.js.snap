// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeightedResultBar should render correctly 1`] = `
<WeightedResultBar
  context="brand"
  count={123}
  percentage={67}
>
  <div
    className="WeightedResultBar"
  >
    <div
      className="WeightedResultBar__details"
    >
      <Text
        context="default"
        inline={true}
        size="normal"
      >
        <span>
          Result
        </span>
      </Text>
      <Text
        context="default"
        inline={true}
        size="normal"
      >
        <span>
          123
        </span>
      </Text>
    </div>
    <ProgressBar
      animated={false}
      context="brand"
      hidden={false}
      percentage={67}
      small={true}
    >
      <div
        aria-hidden={false}
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow={67}
        className="ProgressBar ProgressBar--small"
        role="progressbar"
      >
        <div
          className="ProgressBar__fill ProgressBar__fill--context_brand ProgressBar__fill--small"
          style={
            Object {
              "width": "67%",
            }
          }
        />
      </div>
    </ProgressBar>
  </div>
</WeightedResultBar>
`;
